.github/pull_request_template.md

```diff
@@ -0,0 +1,63 @@
+Fixes #
+
+Resolves #
+
+- [x] Does not depend on other PRs
+- [ ] Depends on other PR being merged first: #
+
+## What kind of change does this PR introduce (check at least one)
+
+- [ ] Bugfix
+- [ ] Improvement
+- [ ] New feature
+- [ ] Refactor
+- [ ] Infrastructure
+- [ ] Other (please specify)
+
+## How would you describe the changes in a changelog (1-2 lines)?
+
+Provide a brief overview of the changes introduced in this pull request.
+
+## Related Issues/PRs (optional)
+
+Link to the relevant issue(s) or provide a clear description of the problem or feature being addressed.
+
+## How to test
+
+- [ ] Needs regression testing (add QA needed label)
+
+## Checklist
+
+- [ ] I have tested these changes locally and they function as expected.
+- [ ] I have added or updated necessary documentation to reflect the changes.
+- [ ] I have followed the project's coding style and guidelines.
+- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
+- [ ] I have checked for any potential conflicts with other branches or pull requests.
+
+## Reviewer Request (optional)
+
+If you have any specific requests or suggestions for reviewers, mention them here.
+
+## Screenshots (optional)
+
+If applicable, provide any additional screenshots or GIFs that showcase the changes made in this pull request.
+
+## Demo Link (optional)
+
+If applicable, provide a link to a live demo or deployment where the changes can be reviewed.
+
+## Known Issues (optional)
+
+If there are any known issues or limitations with the changes made in this pull request, describe them here.
+
+## Additional Notes (optional)
+
+Add any additional notes or comments you think are relevant to this pull request.
+
+## Deploy risk & follow up
+
+Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.
+
+- [x] No risk
+- [ ] Low
+- [ ] High
```

Comment:
I think we can add an optional section that mentions any scripts that may help in testing the PR. It may also contain mentions of automated tests that have been added in this implementation. Just a suggestion, please let me know what you think.

Reply:
I think this is not needed. There are tools like codecov / sonarcloud which does similar thing, if I remember correctly. Although we can add something like this 

```
## Testing and Automated Tests (optional)

Describe any testing scripts, tools, or automated tests that have been added in this implementation to validate the changes in this pull request. If you have included automated test cases, please provide a brief overview of what they cover and how they ensure the correctness of the changes.

```